controllers/utils/multicloud_util.go

```diff
@@ -0,0 +1,341 @@
+		}
+	}
+
+	return mcaForInstallPlan, nil
+}
+
+// NewManagedClusterActionForInstallPlanSpec returns the action spec for approving an InstallPlan.
+func NewManagedClusterActionForInstallPlanSpec(installPlan operatorsv1alpha1.InstallPlan) (actionv1beta1.ActionSpec, error) {
+	installPlanClusterServiceVersionNames, err := json.Marshal(installPlan.Spec.ClusterServiceVersionNames)
+	if err != nil {
+		fmt.Printf("Error trying to Marshal InstallPlan ClusterServiceVersionNames: %s", err.Error())
+	}
+
+	templateContent := fmt.Sprintf(
+		`{"apiVersion": "operators.coreos.com/v1alpha1","kind": "InstallPlan",
+		  "metadata": {"name": "%s","resourceVersion": "%s"},
+		  "spec": {"approval": "Manual","approved": true, "clusterServiceVersionNames": %s}}`,
+		installPlan.ObjectMeta.Name,
+		installPlan.ObjectMeta.ResourceVersion,
+		string(installPlanClusterServiceVersionNames))
+
+	actionSpec := actionv1beta1.ActionSpec{
+		ActionType: actionv1beta1.UpdateActionType,
+		KubeWork: &actionv1beta1.KubeWorkSpec{
+			Resource:       "installplan",
+			Namespace:      installPlan.ObjectMeta.Namespace,
+			ObjectTemplate: runtime.RawExtension{Raw: []byte(templateContent)},
+		},
+	}
+
+	return actionSpec, nil
+}
+
+// CheckManagedClusterActionDone returns error for all the cases in which an action didn't manage to complete
+// successfully.
+func CheckManagedClusterActionDone(
+	ctx context.Context, c client.Client, mcaName string, mcaNamespace string, Log logr.Logger) error {
+	// Check the managedClusterAction has completed successfully.
+	mca := &actionv1beta1.ManagedClusterAction{}
+	err := c.Get(ctx, types.NamespacedName{Name: mcaName, Namespace: mcaNamespace}, mca)
+	if err != nil {
+		return err
+	}
+	// Check that the ManagedClusterAction has completed.
+	conditionMCA := meta.FindStatusCondition(mca.Status.Conditions, actionv1beta1.ConditionActionCompleted)
+	if conditionMCA == nil {
+		err = fmt.Errorf("ManagedClusterAction hasn't completed yet")
+		return err
+
+	}
+	// Check that the ManagedClusterAction has completed successfully.
+	if conditionMCA.Reason != ManagedClusterActionReasonDone {
+		if conditionMCA.Reason == actionv1beta1.ReasonUpdateResourceFailed {
+			err = fmt.Errorf("ManagedClusterAction failed: %s", conditionMCA.Message)
+			return err
+		}
+		err = fmt.Errorf("ManagedClusterAction has unexpected condition reason: %s", conditionMCA.Reason)
+		return err
+	}
+
+	return nil
+}
+
+// DeleteMultiClusterObjects cleans up views associated to an UOCR.
+func DeleteMultiClusterObjects(
+	ctx context.Context, c client.Client, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, Log logr.Logger) error {
+	// Only need to delete ManagedClusterView as ManagedClusterActions get deleted automatically.
+	for _, batch := range clusterGroupUpgrade.Status.RemediationPlan {
+		for _, clusterName := range batch {
+			var mcvLabels = map[string]string{
+				"openshift-cluster-group-upgrades/clusterGroupUpgrade": clusterGroupUpgrade.Name}
```

Comment:
If we have two CGU in different namespaces working with the same cluster then we could end up with one of the CGU deleting the multicluster objects from the other. If we add the CGU namespace to the label we would avoid any chance this happens.